tracer/src/Datadog.Trace/Configuration/ConfigurationSources/JsonConfigurationSource.cs

```diff
@@ -344,7 +344,7 @@ bool ITelemeteredConfigurationSource.IsPresent(string key)
 
             try
             {
-                var valueAsString = token?.Value<string>();
+                var valueAsString = token?.ToString(Formatting.None);
```

Comment:
Maybe this is better?
```suggestion
                var valueAsString = token?.Type is JTokenType.Null or JTokenType.None or JTokenType.Undefined or null
                                        ? null
                                        : token.Value<string>(Formatting.None);
```

Reply:
Good catch, and fixed (with nested `switch` expressions). This wasn't working as expected and was causing many tests to fail.